src/MediatR/AsyncRequestHandler.cs

```diff
@@ -11,7 +11,7 @@ public abstract class AsyncRequestHandler<TMessage> : IAsyncRequestHandler<TMess
     {
         public async Task<Unit> Handle(TMessage message)
         {
-            await HandleCore(message);
+            await HandleCore(message).ConfigureAwait(false);
```

Comment:
And to answer your question - yes, people would want to access things with context, inside that internal HandleCore method.